v-next/hardhat/src/internal/helpers/config-loading.ts

```diff
@@ -0,0 +1,55 @@
+import { isAbsolute, resolve } from "node:path";
+
+import { findUp } from "@nomicfoundation/hardhat-utils/fs";
+
+async function findClosestHardhatConfig(): Promise<string> {
+  let hardhatConfigPath = await findUp("hardhat.config.js");
+
+  if (hardhatConfigPath !== undefined) {
+    return hardhatConfigPath;
+  }
+
+  hardhatConfigPath = await findUp("hardhat.config.ts");
```

Comment:
I think `.ts` should be before `.js`. So that if a user accidentally compiles his config once, we still use the most updated `.ts` version.